src/apps/natdiscovery/natdiscovery.c

```diff
@@ -356,14 +356,14 @@ static int init_socket(int *socketfd, ioa_addr *local_addr, int local_port, ioa_
   return ret;
 }
 
-static int stunclient_send(stun_buffer *buf, int sockfd, ioa_addr *local_addr, int *local_port, ioa_addr *remote_addr,
-                           int change_ip, int change_port, int padding, int response_port) {
+static int stunclient_send(stun_buffer *buf, int sockfd, ioa_addr *local_addr, uint16_t *local_port, ioa_addr *remote_addr,
+                           int change_ip, uint16_t change_port, int padding, uint16_t response_port) {
   int ret = 0;
 
   stun_prepare_binding_request(buf);
 
   if (response_port >= 0) {
```

Comment:
this function is another place where you have to evaluate all the places where `local_port` is used, and adjust any if-checks that are now always true.